src/workerd/io/worker.h

```diff
@@ -362,40 +363,9 @@ class Worker::Isolate: public kj::AtomicRefcounted {
   // after the refcount reaches 0, `tryAddStrongRef` is always still safe to invoke even if the
   // underlying Isolate memory has been deallocated (provided ownership of the weak isolate
   // reference is retained).
-  class WeakIsolateRef: public kj::AtomicRefcounted {
-    // TODO(someday): This can be templatized & even integrated into KJ. That's why the method
-    // bodies are defined inline.
-  public:
-    WeakIsolateRef(Isolate* thisArg) : this_(thisArg) {}
-
-    // This tries to materialize a strong reference to the isolate. It will fail if the isolate's
-    // refcount has already dropped to 0. As discussed in the class, the lifetime of this weak
-    // reference can exceed the lifetime of the isolate it's tracking.
-    kj::Maybe<kj::Own<const Worker::Isolate>> tryAddStrongRef() const {
-      auto lock = this_.lockShared();
-      if (*lock == nullptr) {
-        return kj::none;
-      }
-
-      return kj::atomicAddRefWeak(**lock);
-    }
-
-    // This is invoked by the Isolate destructor to clear the pointer. That means that any racing
-    // code will never try to invoke `atomicAddRefWeak` on the instance any more. Any code racing
-    // in between the refcount dropping to 0 and the invalidation getting invoked will still fail
-    // to acquire a strong reference. Any code acquiring a strong reference prior to the refcount
-    // dropping to 0 will prevent invalidation until that extra reference is dropped.
-    void invalidate() const {
-      *this_.lockExclusive() = nullptr;
-    }
+  using WeakIsolateRef = AtomicWeakRef<Isolate>;
```

Comment:
We should move all or parts of the comment now above this line to weak-refs.h to describe the `AtomicWeakRef` class, otherwise the comment at weak-refs line 18 (`... As discussed in the class...`) becomes difficult to understand.